dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -180,91 +189,67 @@ def _check_dataset_time_consistency_by_time_array(self, time_dim, load_data_df):
-            # map and consolidate from_fraction only
-            records = models_to_dataframe(records).filter("to_id IS NOT NULL")
-            df = (
-                df.join(records, on=df[column] == records.from_id, how="inner")
-                .drop("from_id")
-                .drop(column)
-                .withColumnRenamed("to_id", column)
-            ).filter(f"{column} IS NOT NULL")
-            nonfraction_cols = [x for x in df.columns if x not in {"fraction", "from_fraction"}]
-            df = df.fillna(1, subset=["from_fraction"]).selectExpr(
-                *nonfraction_cols, "fraction*from_fraction AS fraction"
-            )
-
-            # After remapping, rows in load_data_lookup for standard_handler and rows in load_data for one_table_handler may not be unique;
-            # imagine 5 subsectors being remapped/consolidated to 2 subsectors.
-
-        # if mapping is for pivoted dimension and df contains pivoted dimension cols
-        elif column == self.get_pivot_dimension_type().value and not set(
-            self.get_pivot_dimension_columns()
-        ).difference(set(df.columns)):
-            # map and consolidate pivoted dimension columns (need pytest)
-            # this is only for dataset_schema_handler_one_table,
-            # b/c handlder_standard._remap_dimension_columns() only takes in load_data_lookup
-            nonvalue_cols = list(
-                set(df.columns).difference(set(self.get_pivot_dimension_columns()))
-            )
-
-            records_dict = defaultdict(dict)
-            for row in records:
-                if row.to_id is not None:
-                    records_dict[row.to_id][row.from_id] = row.from_fraction
-
-            to_ids = sorted(records_dict)
-            value_operations = []
-            for tid in to_ids:
-                operation = "+".join(
-                    [f"{from_id}*{fraction}" for from_id, fraction in records_dict[tid].items()]
-                )  # assumes reduce by summation
-                operation += f" AS {tid}"
-                value_operations.append(operation)
-
-            df = df.selectExpr(*nonvalue_cols, *value_operations)
-
-        else:
-            pass
+            records = mapping_config.get_records_dataframe()
+            if column in df.columns:
+                df = map_and_reduce_non_pivot_dimension(df, records, column)
+            elif column == self.get_pivot_dimension_type().value and not pivot_columns.difference(
+                df.columns
+            ):
+                # TODO: Do we want operation to be configurable?
+                operation = "sum"
+                df, _ = map_and_reduce_pivot_dimension(
+                    df,
+                    records,
+                    pivot_columns,
+                    operation,
+                    rename=False,
+                )
+            elif column == self.get_pivot_dimension_type().value and pivot_columns.intersection(
```

Comment:
This should be `column == self.get_pivot_dimension_type().value and not pivot_columns.intersection(df.columns)`, no exception, just passing.

Because if the mapping is about a pivoted dimension but the df does not have the pivoted dim, then it does nothing.

Reply:
Keep as-is because this will catch a bug if it happens.